test/account_multisig.test.js

```diff
@@ -0,0 +1,126 @@
+        onResult: async () => {
+            const { requestId } = accountMultisig.getRequest();
+            console.log('requestId', requestId);
+            // set confirmKey as signer
+            const originalSigner = nearjs.connection.signer;
+            const tempKeyStore = new InMemoryKeyStore();
+            await tempKeyStore.setKey(nearjs.connection.networkId, accountId, accountMultisig.confirmKey);
+            nearjs.connection.signer = new InMemorySigner(tempKeyStore);
+            // 2nd confirmation signing with confirmKey from Account instance
+            await account.signAndSendTransaction(accountId, [
+                functionCall('confirm', { request_id: requestId }, MULTISIG_GAS, MULTISIG_DEPOSIT)
+            ]);
+            nearjs.connection.signer = originalSigner;
+        }
+    });
+    accountMultisig.confirmKey = KeyPair.fromRandom('ed25519');
+    accountMultisig.postSignedJson = () => ({ publicKey: accountMultisig.confirmKey.getPublicKey() });
+    accountMultisig.getRecoveryMethods = () => ({
+        data: keys.map(keyMapping)
+    });
+    try {
+        await accountMultisig.deployMultisig([...fs.readFileSync('./test/data/multisig.wasm')]);
+    } catch (e) {
+        console.log(e);
+        if (e.message.indexOf('Contract method is not found') === -1) {
+            throw(e);
+        }
+    }
+    return accountMultisig;
+};
+
+beforeAll(async () => {
+    nearjs = await testUtils.setUpTestConnection();
+    let nodeStatus = await nearjs.connection.provider.status();
+    startFromVersion = (version) => semver.gte(nodeStatus.version.version, version);
+    console.log(startFromVersion);
+});
+
+describe('deployMultisig key rotations', () => {
+
+    test('full access key if recovery method is "ledger" or "phrase"', async () => {
+        const account = await testUtils.createAccount(nearjs);
+        await account.addKey(KeyPair.fromRandom('ed25519').getPublicKey());
+        const keys = await account.getAccessKeys();
+        const accountMultisig = await AccountMultisigInstance(
+            account,
+            ({ public_key: publicKey }, i) => ({ publicKey, kind: i > 0 ? 'ledger' : 'phrase' })
+        );
+        const currentKeys = await accountMultisig.getAccessKeys();
+        expect(currentKeys.find(({ public_key }) => keys[0].public_key === public_key).access_key.permission).toEqual('FullAccess');
+        expect(currentKeys.find(({ public_key }) => keys[1].public_key === public_key).access_key.permission).toEqual('FullAccess');
+    });
+
+    test('access key if method is "phone"', async () => {
+        const account = await testUtils.createAccount(nearjs);
+        const keys = await account.getAccessKeys();
+        const accountMultisig = await AccountMultisigInstance(account);
+        const currentKeys = await accountMultisig.getAccessKeys();
+        expect(currentKeys.find(({ public_key }) => keys[0].public_key === public_key).access_key.permission).not.toEqual('FullAccess');
```

Comment:
I think it's actually best to join these 2 tests so that you check that decision is made per each key and not for account as a whole.